index.js

```diff
@@ -51,25 +28,62 @@ exports.promptPrefix = '';
 exports.debugFlag = '--debug';
 
 /**
- * Boot mage
+ * Force-configure mage to use cluster: 1 (one master, one process),
+ * and set the list of files and folders to watch for.
+ */
+var config = require('../mage/lib/config');
+config.set('server.cluster', 1);
+
+var WATCH_FILES = config.get('external.mage-console.watch', []);
+
+WATCH_FILES.push(APP_CONFIG_PATH);
+WATCH_FILES.push(APP_LIB_PATH);
+
+/**
+ * Load the application and boot
  */
 var mage = require(APP_LIB_PATH);
+var logger = mage.core.logger.context('REPL');
 var processManager = mage.core.processManager;
 
 mage.boot();
 
-var clusterConfiguration = mage.core.config.get('server.cluster');
+function setRawMode(val) {
+	var stdin = process.stdin;
+
+	if (!stdin.setRawMode) {
+		logger
+			.emergency
+			.details('This may happen when using terminals such as MINGW64 on Windows')
+			.details('Please try to use PowerShell or cmd.exe instead')
+			.log('Cannot run mage-console; cannot switch to raw mode');
 
-if (clusterConfiguration !== 1) {
-	console.error('');
-	console.error('mage-console requires your application to be configured with');
-	console.error('"server.cluster" to 1. Please change your configuration and try again');
-	console.error('');
+		mage.quit(1, true);
+	}
 
-	process.exit(-1);
+	stdin.setRawMode(val);
 }
 
-var logger = mage.core.logger.context('REPL');
+function onceSomeFilesChanged(onChange) {
+	var called = false;
+	var watcher = watch(WATCH_FILES, {
+		recursive: true
+	}, function (event, name) {
+		if (name.split(path.sep).pop()[0] === '.') {
```

Comment:
I know this code existed before, but what is this about? Hidden files? Why ignore them exactly? (.DS_Store and stuff?)

Reply:
Correct. IIRC WebStorm and some other IDEs create and update dot files on file save; this was a quick fix to ensure file watch woild not trigger more than once (arguably not a real fix, but in this case it got the job done).